features/steps/dataset_feature.go

```diff
@@ -0,0 +1,240 @@
+package steps_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/ONSdigital/dp-dataset-api/config"
+	"github.com/ONSdigital/dp-dataset-api/models"
+	"github.com/ONSdigital/dp-dataset-api/mongo"
+	"github.com/ONSdigital/dp-dataset-api/service"
+	serviceMock "github.com/ONSdigital/dp-dataset-api/service/mock"
+	"github.com/ONSdigital/dp-dataset-api/store"
+	storeMock "github.com/ONSdigital/dp-dataset-api/store/datastoretest"
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	kafka "github.com/ONSdigital/dp-kafka/v2"
+	"github.com/ONSdigital/dp-kafka/v2/kafkatest"
+	featuretest "github.com/armakuni/dp-go-featuretest"
+	"github.com/benweissmann/memongo"
+	"github.com/cucumber/godog"
+	"github.com/globalsign/mgo"
+	"github.com/maxcnunes/httpfake"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+type DatasetFeature struct {
+	ErrorFeature    featuretest.ErrorFeature
+	svc             *service.Service
+	errorChan       chan error
+	Datasets        []*models.Dataset
+	MongoClient     *mongo.Mongo
+	Config          *config.Configuration
+	HTTPServer      *http.Server
+	FakeAuthService *httpfake.HTTPFake
+	ServiceRunning  bool
+}
+
+func NewDatasetFeature(mongoCapability *featuretest.MongoCapability) *DatasetFeature {
+
+	f := &DatasetFeature{
+		HTTPServer:      &http.Server{},
+		errorChan:       make(chan error),
+		Datasets:        make([]*models.Dataset, 0),
+		FakeAuthService: httpfake.New(),
+		ServiceRunning:  false,
+	}
+
+	var err error
+
+	f.Config, err = config.Get()
+	if err != nil {
+		panic(err)
```

Comment:
Replace `panic(err)` with `return nil, err` and handle error in `InitializeScenario`